plugin/trino-hive/src/main/java/io/trino/plugin/hive/parquet/ParquetTypeTranslator.java

```diff
@@ -49,6 +51,9 @@ private ParquetTypeTranslator() {}
             }
         }
         if (toTrinoType instanceof VarcharType varcharType) {
+            if (fromParquetType == DOUBLE) {
+                return Optional.of(new DoubleToVarcharCoercer(varcharType, false));
```

Comment:
Can we just make a different class for orc OrcDoubleToVarcharCoercer ? I think it's better to let there be some code repetition than to have that `isOrcFile` check for every position